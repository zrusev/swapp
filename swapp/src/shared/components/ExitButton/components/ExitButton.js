import styled from 'styled-components';
import logo from '../assets/logout.png';

export const ExitButton = styled.div`
  width: 25px;
  height: 25px;
  background-image: url(${logo});
  background-position: center;
  background-repeat: no-repeat;
  padding: 0.8em 1em;

  &:hover,
  &:focus {
    box-shadow: rgba(255, 255, 255, 0.45) 0px 0px 10px 0px;
  }
`;
