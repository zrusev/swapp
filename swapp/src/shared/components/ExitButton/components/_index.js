export * from './ExitContainer';
export * from './ExitButton';
