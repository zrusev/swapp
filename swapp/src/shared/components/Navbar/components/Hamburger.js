import styled from 'styled-components';

export const Hamburger = styled.button`
    display: none;
    grid-area: burger;
    margin: 0 20px 0 0;
    padding: 0;
    justify-self: end;
    font-size: 40px;
    border: none;
    background: none;
    outline: none;
    transition: 0.1s;
    
    &:active {
        transform: scale(1.2);
    }

    @media (max-width: 700px) {
        display: inline;
    }
`;