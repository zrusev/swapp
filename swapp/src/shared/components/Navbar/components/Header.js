import styled from 'styled-components';

export const Header = styled.header`
  overflow: hidden;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: auto;
  z-index: 1;
  display: grid;
  grid-template-areas: 'logo logo nav';
  border-color: ${({ theme: { appBarBorderColor } }) => appBarBorderColor};
  background: ${({ theme: { appBarBackground } }) => appBarBackground};

  @media (max-width: 700px) {
    grid-template-areas: 'logo logo burger' 'nav nav nav';
  }
`;
