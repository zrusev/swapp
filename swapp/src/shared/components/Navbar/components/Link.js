import styled from 'styled-components';
import { Link as RouterLink} from 'react-router-dom';

export const Link = styled(RouterLink)`
    margin: 0.5em 0;
    padding: 0.5em 0.5em;
    color: ${({theme: {appBarFontColor}}) => appBarFontColor};
    text-decoration: none;
    font-size: 0.8em;

    &:hover,
    &:focus {
        box-shadow: rgba(255, 255, 255, 0.45) 0px 0px 10px 0px;
    }
`;