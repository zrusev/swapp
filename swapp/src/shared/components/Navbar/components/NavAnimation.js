import styled from 'styled-components';

export const NavAnimation = styled.div`
  @media (max-width: 700px) {
    grid-column: 1 / 4;
  }

  &.NavAnimation-enter {
    opacity: 0;
    transform: scale(0.5);
  }

  &.NavAnimation-enter-active {
    opacity: 1;
    transform: translateX(0);
    transition:
      opacity 350ms,
      transform 350ms;
  }

  &.NavAnimation-exit {
    opacity: 1;
  }

  &.NavAnimation-exit-active {
    opacity: 0;
    transform: scale(0.5);
    transition:
      opacity 350ms,
      transform 350ms;
  }
`;
