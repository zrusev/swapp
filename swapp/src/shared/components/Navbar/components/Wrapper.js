import styled from 'styled-components';

export const Wrapper = styled.div`
    display: grid;
    grid-area: nav;
    grid-template-columns: repeat(4, auto);
    align-items: center;
    justify-items: flex-end;
    border-color: ${({theme: {appBarBorderColor}}) => appBarBorderColor};
    background: ${({theme: {appBarBackground}}) => appBarBackground};

    a:hover {
        transform: scale(1.1);
    }

    @media (max-width: 700px) {
        grid-template-rows: repeat(4, auto);
        grid-template-columns: none;
        grid-row-gap: 20px;
    
        background: rgba(40, 44, 47, 0.95);
        box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);

        display: flex;
        flex-direction: column;
    }
`;