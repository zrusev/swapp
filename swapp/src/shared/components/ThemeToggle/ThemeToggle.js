import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Logo } from './components/_index';
import { ThemeContext } from '../../../style/contexts/themeContext';

const ThemeToggle = ({ children }) => {
  const { toggleTheme } = useContext(ThemeContext);

  return <Logo onClick={toggleTheme}>{children}</Logo>;
};

ThemeToggle.propTypes = {
  children: PropTypes.object.isRequired,
};

export default ThemeToggle;
