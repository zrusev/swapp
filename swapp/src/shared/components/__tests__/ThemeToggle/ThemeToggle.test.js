import React from 'react';
import { shallow, mount } from 'enzyme';

import ThemeToggle from '../../ThemeToggle/ThemeToggle';
import { Logo } from '../../ThemeToggle/components/_index';
import { ThemeContext } from '../../../../style/contexts/themeContext';

describe('should ThemeToggle acts as a function container', () => {
  const props = {
    toggleTheme: jest.fn(),
  };

  it('should match description', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={{ ...props }}>
        <ThemeToggle>
          <span>Click here</span>
        </ThemeToggle>
      </ThemeContext.Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should respond on click action', () => {
    const wrapper = mount(
      <ThemeContext.Provider value={{ ...props }}>
        <ThemeToggle>
          <span>Click here</span>
        </ThemeToggle>
      </ThemeContext.Provider>,
    );

    wrapper.find(Logo).simulate('click');
    expect(props.toggleTheme).toHaveBeenCalled();
  });
});
