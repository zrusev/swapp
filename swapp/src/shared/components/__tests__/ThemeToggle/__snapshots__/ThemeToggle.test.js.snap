// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should ThemeToggle acts as a function container should match description 1`] = `
<ThemeToggle>
  <span>
    Click here
  </span>
</ThemeToggle>
`;
